understanding/21/non-text-contrast.html

```diff
@@ -29,6 +29,16 @@ <h2>Intent</h2>
 				<p>The 3:1 contrast ratios referenced in this Success Criterion is intended to be treated as threshold values. When comparing the computed contrast ratio to the Success Criterion ratio, the computed values should not be rounded (e.g. 2.999:1 would not meet the 3:1 threshold).</p>
 			</div>
 
+			<div class="note">
+				<p>User agents which render non-text content may apply anti-aliasing, shadow blending, and other techniques which can cause some pixels of the content to render in fainter colors than those specified by authors. When an author uses markup or stylesheets to specify that a particular bit of content should be a specific color, as long as the some area of the content reaches the author-specified color, evaluators may refer to the foreground and background colors obtained from the user agent, or the underlying markup and stylesheets, rather than evaluating individual pixels within the text as presented on screen.</p>
+
+				<p>For example, if an author uses CSS to specify that a circular button's border should be a specific shade of blue, evaluators may refer to the markup-specified color even though most user agents will apply anti-aliasing when rendering a circular button, as long as some area of the border is actually rendered at that specific color.</p>
+
+				<p>However, some authoring techniques can result in components or graphics which user agents will render using much fainter colors than the colors defined in underlying markup and stylesheets. This can lead to situations where a user agent reports that content has a contrast ratio that nominally passes the Success Criterion, but the content as-rendered has a much lower contrast in practice and fails the Success Criterion.</p>
+
+				<p>In these cases, best practice would be for authors to choose an technique which guarantees that the user agent will render the content with a contrast ratio that meets the Success Criterion. For example, an author seeking to use a line drawn around a button to meet this Success Criterion might choose to use a CSS <code>border</code> or <code>outline</code> property, rather than a CSS <code>box-shadow</code> property; most user agents will attempt to render fractional pixels of a shadow using fainter colors, but will not use this technique to render fractional pixels of a border or outline.</p>
```

Comment:
```suggestion
				<p>In these cases it is necessary for authors to choose an technique which guarantees that the user agent will render the content with a contrast ratio that meets the Success Criterion. For example, an author seeking to use a line drawn around a button to meet this Success Criterion might choose to use a CSS <code>border</code> or <code>outline</code> property, rather than a CSS <code>box-shadow</code> property; most user agents will attempt to render fractional pixels of a shadow using fainter colors, but will not use this technique to render fractional pixels of a border or outline.</p>
```

Reply:
as authors have no absolute control over the myriad of possible reasons why something gets antialiased too heavily, i feel very uneasy about making such hard requirements for binary pass/fail here